CONTINUATION.md

```diff
@@ -0,0 +1,511 @@
+instance `p2` with a string of `'error message'`. In this case, the event is
+scheduled in a new microtask which was scheduled in the context of `'reject'`.
+
+For the two type of context variables, the value at site (2) with
+`unhandledrejection` event of `p2`:
+
+- For `AsyncContext.Variable`, the context is `'init'`, where `p2`
+  attaches the promise handlers to `p1`.
+- For `ContinuationFlow`, the context is `'reject'`, where `p1` was rejected.
+
+---
+
+If handlers were attached to an already rejected promise:
+
+```js
+let p1 // (1)
+ctx.run('reject', () => {
+  p1 = Promise.reject('error message')
+})
+
+ctx.run('init', () => {
+  const p2 = p1 // p1 is already rejected
+    .then(undefined, undefined) // (2)
+})
+
+addEventListener('unhandledrejection', (event) => {
+  ctx.get() // (2)
+})
+```
+
+In this case, the `unhandledrejection` event is scheduled in the context where
+`.then` is called, that is `'init'`.
+
+> By saying "the `unhandledrejection` event is scheduled" above, it is
+> referring to [HostPromiseRejectionTracker](https://html.spec.whatwg.org/#the-hostpromiserejectiontracker-implementation).
+> This host hook didn't actually "schedule" the event dispatching, rather putting the
+> promise in a queue and the event is actually scheduled from [event loop](https://html.spec.whatwg.org/#notify-about-rejected-promises).
+
+For the two type of context variables, the value at site (2) with
+`unhandledrejection` event of `p2`:
+
+- For `AsyncContext.Variable`, the context is `'init'`, where `p2`
+  attaches the promise handlers to `p1`.
+- For `ContinuationFlow`, the context is `'reject'`, where `p1` was rejected.
+
+### Fulfilled promises
+
+Similar to the rejected promise issue, the MOST relevant cause's context when
+scheduling a microtask for newly created promise handlers are not
+deterministic:
+
+```js
+let p1 // (1)
+ctx.run('resolve', () => {
+  p1 = Promise.resolve('yay')
+})
+
+ctx.run('init', () => {
+  const p2 = p1 // p1 is already resolved
+    .then(() => {
+      ctx.get() // (2)
+    })
+})
+```
+
+[`PerformPromiseThen`](https://tc39.es/ecma262/#sec-performpromisethen) calls
+[`HostEnqueuePromiseJob`](https://html.spec.whatwg.org/#hostenqueuepromisejob),
+which immediately queues a new microtask to call the promise fulfill reaction.
+
+Explaining in the callback continuation form, this would be:
+
+```js
+// `Promise.prototype.then` in plain JS.
+const then = (p, onFulfill) => {
+  if (p[PromiseState] === 'FULFILLED') {
+    let { resolve, reject, promise } = Promise.withResolvers()
+    queueMicrotask(() => {
+      resolve(onFulfill(promise[PromiseResult]))
+    })
+    return promise
+  }
+  // ...
+}
+let p1 // (1)
+ctx.run('resolve', () => {
+  p1 = Promise.resolve('yay')
+})
+
+ctx.run('init', () => {
+  const p2 = then(p1, undefined) // the promise is already resolved
+})
+```
+
+In this case, the MOST relevant context would be `'init'` since in this
```

Comment:
Most relevant context for what?  For awaiting `p2`?  It would be clearer if you put an actual `// (1)` somewhere to indicate which context you mean.